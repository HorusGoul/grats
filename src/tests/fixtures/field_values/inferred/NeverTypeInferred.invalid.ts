/** @gqlType */
export default class Query {
  /** @gqlField */
  foo(): never {
    throw new Error("Not implemented");
  }
}
