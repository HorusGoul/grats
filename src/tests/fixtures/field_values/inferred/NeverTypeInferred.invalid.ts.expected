-----------------
INPUT
----------------- 
/** @gqlType */
export default class Query {
  /** @gqlField */
  foo(): never {
    throw new Error("Not implemented");
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/field_values/inferred/NeverTypeInferred.invalid.ts:4:10 - error: Could not infer the GraphQL type. TypeScript sees this value as being typed as `never`. Perhaps you missing a type annotation, or have a type error elsewhere in your code?.

If you think Grats should be able to infer this type, please report an issue at https://github.com/captbaritone/grats/issues.

4   foo(): never {
           ~~~~~
